gen/enum_test.go

```diff
@@ -399,3 +400,107 @@ func TestEnumAccessors(t *testing.T) {
 		})
 	})
 }
+
+func TestEnumLabelLegit(t *testing.T) {
+	tests := []struct {
+		enumItem  te.EnumWithLabel
+		wireValue string
+	}{
+		{te.EnumWithLabelUsername, "\"surname\""},
+		{te.EnumWithLabelPassword, "\"hashed_password\""},
+		{te.EnumWithLabelSalt, "\"salt\""},
+		{te.EnumWithLabelSugar, "\"sugar\""},
```

Comment:
We should add a case for unknown enums here. (They're still valid.)

    {te.EnumWithLabel(42), "42"}